api/internal/core/storage/adminapi.go

```diff
@@ -0,0 +1,190 @@
+
+// NewAdminAPIStorage will create an instance of AdminAPIStorage and complete the basic configure
+func NewAdminAPIStorage() *AdminAPIStorage {
+	return &AdminAPIStorage{
+		client: resty.New().
+			SetBaseURL("http://127.0.0.1:9080/apisix/admin/").
+			SetHeader("X-API-KEY", "edd1c9f034335f136f87ad84b625c8f1").
+			SetTimeout(1 * time.Second),
+	}
+}
+
+// AdminAPIStorage is the storage layer implementation of the APISIX Admin API
+type AdminAPIStorage struct {
+	client *resty.Client
+}
+
+func (s AdminAPIStorage) Get(ctx context.Context, key string) (string, error) {
+	resp, err := s.client.R().
+		SetContext(ctx).
+		Get(s.removeKeyPrefix(key))
+	if err != nil {
+		return "", errors.Wrap(err, adminAPIErrorWarp)
+	}
+
+	json := gjson.Parse(resp.String())
+	if json.Get("count").Int() < 1 {
+		return "", errors.Errorf("key: %s is not found", key)
+	}
+
+	return gjson.Get(resp.String(), "node.value").Raw, nil
+}
+
+func (s AdminAPIStorage) List(ctx context.Context, key string) ([]storage_api.Keypair, error) {
+	resp, err := s.client.R().
+		SetContext(ctx).
+		Get(s.removeKeyPrefix(key))
+	if err != nil {
+		return nil, errors.Wrap(err, adminAPIErrorWarp)
+	}
+
+	switch resp.Header().Get("X-API-VERSION") {
+	case "v3":
+		return s.parseList(adminApiVersion3, resp.String()), nil
+	case "v2":
+		fallthrough
+	default:
+		return s.parseList(adminAPIVersion2, resp.String()), nil
+	}
+}
+
+func (s AdminAPIStorage) Create(ctx context.Context, key, val string) error {
+	return s.Update(ctx, key, val)
+}
+
+func (s AdminAPIStorage) Update(ctx context.Context, key, val string) error {
+	resp, err := s.client.R().
+		SetContext(ctx).
+		SetHeader("Content-Type", "application/json").
+		SetBody(val).
+		Put(s.removeKeyPrefix(key))
+	if err != nil {
+		return errors.Wrap(err, adminAPIErrorWarp)
+	}
+
+	if err = s.checkError(resp.String()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s AdminAPIStorage) BatchDelete(ctx context.Context, keys []string) error {
+	for _, key := range keys {
+		resp, err := s.client.R().
+			SetContext(ctx).
+			Delete(s.removeKeyPrefix(key))
+		if err != nil {
+			return errors.Wrap(err, adminAPIErrorWarp)
+		}
+
+		if err = s.checkError(resp.String()); err != nil {
+			return fmt.Errorf("delete key[%s] failed: %s", key, err)
+		}
+
+		if gjson.Get(resp.String(), "deleted").Int() < 1 {
+			return errors.Errorf("key: %s is not found", key)
+		}
+	}
+
+	return nil
+}
+
+// Watch watches for changes in data sources and generates event streams
+// Admin API storage implementation does not support Watch capability
+func (s AdminAPIStorage) Watch(_ context.Context, _ string) <-chan storage_api.WatchResponse {
+	return make(chan storage_api.WatchResponse)
+}
+
+// removeKeyPrefix will remove the "/apisix/" prefix of key
+// the pattern of the key is usually like "/apisix/routes"
+func (AdminAPIStorage) removeKeyPrefix(s string) string {
+	return strings.ReplaceAll(s, "/apisix/", "")
```

Comment:
the prefix needs to be configurated too.

Reply:
Hi, @nic-chen 

Yes, same with https://github.com/apache/apisix-dashboard/pull/2570#discussion_r939836946, it will be finish in next PR, we only use a default value in this PR.